router/router.go

```diff
@@ -725,44 +730,89 @@ func (worker *workerT) handleWorkerDestinationJobs(ctx context.Context) {
 			}
 		}
 
-		var sourceIDs []string
-		for i, destinationJobMetadata := range destinationJob.JobMetadataArray {
+		for _, destinationJobMetadata := range destinationJob.JobMetadataArray {
 			handledJobMetadatas[destinationJobMetadata.JobID] = &destinationJobMetadata
 
-			attemptNum := destinationJobMetadata.AttemptNum
-			if attemptedToSendTheJob {
-				attemptNum++
-			}
+			_destinationJob := destinationJob
+			_destinationJobMetadata := destinationJobMetadata
 
-			status := jobsdb.JobStatusT{
-				JobID:         destinationJobMetadata.JobID,
-				ExecTime:      time.Now(),
-				RetryTime:     time.Now(),
-				AttemptNum:    attemptNum,
-				ErrorCode:     strconv.Itoa(respStatusCode),
-				ErrorResponse: []byte(`{}`),
-				Parameters:    []byte(`{}`),
+			routerJobResponses = append(routerJobResponses, &RouterJobResponse{jobID: destinationJobMetadata.JobID,
+				destinationJob:         &_destinationJob,
+				destinationJobMetadata: &_destinationJobMetadata,
+				respStatusCode:         respStatusCode,
+				respBody:               respBody,
+				attemptedToSendTheJob:  attemptedToSendTheJob})
+		}
+	}
+
+	sort.Slice(routerJobResponses, func(i, j int) bool {
+		return routerJobResponses[i].jobID < routerJobResponses[j].jobID
+	})
+
+	destLiveEventSentMap := make(map[*types.DestinationJobT]struct{})
+
+	for _, routerJobResponse := range routerJobResponses {
+		destinationJobMetadata := routerJobResponse.destinationJobMetadata
+		destinationJob := routerJobResponse.destinationJob
+		attemptedToSendTheJob := routerJobResponse.attemptedToSendTheJob
+		if !isSuccessStatus(respStatusCode) {
+			if prevFailedJobID, ok := userToJobIDMap[destinationJobMetadata.UserID]; ok {
+				//This means more than two jobs of the same user are in the batch & the batch job is failed
+				//Only one job is marked failed and the rest are marked waiting
+				//Job order logic requires that at any point of time, we should have only one failed job per user
+				//This is introduced to ensure the above statement
+				resp := fmt.Sprintf(`{"blocking_id":"%v", "user_id":"%s", "moreinfo": "attempted to send in a batch"}`, prevFailedJobID, destinationJobMetadata.UserID)
```

Comment:
`%d` is safer it this case. `%v` can break json if a different field is provided
```suggestion
				resp := fmt.Sprintf(`{"blocking_id":"%d", "user_id":"%s", "moreinfo": "attempted to send in a batch"}`, prevFailedJobID, destinationJobMetadata.UserID)
```